### DISCUSSION TITLE
Vercel Failing to Generate Static Pages at Build Time with NextJS 14.1 Pages Router (getStaticProps)

### DISCUSSION THREAD

**Discussion by fluencyforge** [vercel/community#5919] [Help]
### Summary

A few days ago, we started noticing that Vercel seems to no longer be able to successfully build static pages leveraging NextJS's getStaticProps / getStaticPaths method from the Pages Router. For many months, we have used the same code to generate static pages at build time for our project on Vercel, but we recently noticed our builds show a client-side error when trying to view those pages in the browser. Looking in the console, it seems that the error is occurring because the props are not successfully being passed to the page components at build time, which means all the props are undefined in the final page components as a result. What is really odd about this is that the pages are being shown as generated in the Vercel build logs and the build is actually deploying successfully (the static page routes exist in the builds too), so it hasn't been clear how to debug this issue.

Here's what we have encountered and why we are convinced this is actually an issue with Vercel and not the code we have:

1. We are fetching data from an API to pass in as props to the static pages we are generating, so we initially thought there was a breakdown there causing the problem. To check, we placed a console log, which printed the props being passed from getStaticProps, to view on the Vercel build log (all of the API data was logging out as expected in the build log). We even went as far as removing the API from the equation and hard-coding the data getStaticProps will pass to all of the pages, and it still resulted in the same issue post-build on Vercel.

2. We have tested local builds of the project, and the pages are getting generated as expected.

3. If we deploy the same project, with the exact same code, to **Netlify** the build works as expected and all of the pages are generated correctly. We can view all of the static pages normally. This is the most concerning observation we had.

4. When we try to access the generated pages in the Vercel deployment, a file is oddly automatically downloading from the page that has all of the data that was supposed to be used for the props in the page's components and a bunch of other stuff I am guessing is leftover from the build process.

Having seen all of that, it really seems like Vercel is somehow not successfully passing the props correctly to populate the static pages at build time. We have been looking for evidence that Vercel is no longer supporting the Pages Router methods, but that doesn't seem to be the case. Even more concerning is no one else seems to have reported this issue that we can find, because we would expect others to be encountering the same problem on Vercel, given we are very recently seeing this behavior on Vercel specifically, and it is completely failing to generate static pages with NextJS with this common approach. As much as we'd like to avoid it, we are looking at potentially converting that entire part of our application to the App Router to see if we can overcome this issue that way.

Any thoughts or insights into this issue would be very much appreciated. This is a major setback for us to have to rethink this part of our project, so we are trying to find the best way to handle it at the moment.

Thanks!

### Steps to Reproduce

Attempt to generate static pages on Vercel with NextJS 14.1 and the Pages Router using getStaticProps / getStaticPaths. If the issue occurs, props will not be provided to the components within the static pages, which will result in undefined props in the build. Given the code we have is working everywhere else outside of Vercel, we believe Vercel is currently having an issue doing this.

**Comment by leerob** [comment]
Could you share a minimal reproduction? We are using `getStaticProps` for Vercel's own website in production on the latest version and have not seen this.

**Reply by fluencyforge** [reply]
Hey Lee, I really appreciate you responding to this. We were trying to recreate the issue in a sandbox for you to test, but we haven't been able to reproduce it. One action we just took was to have one of the developers on my team attempt to deploy our project through his personal Vercel account, and incredibly the pages built on his deployment of the project correctly. After seeing the result he had, I made a 2nd project in my Vercel account, pointing to the same code in our repo, and I am still seeing the issues with the new project deployment from my account. At this point, I think I will have to subscribe to a Vercel Pro account to get to the bottom of this. The problem seems to be related to my Vercel account specifically / the deployments coming out of my account. We are getting ready to officially productionalize our application, so I was going to upgrade to a Vercel Pro account at that time anyway.

**Reply by fluencyforge** [reply]
We ended up transferring the project to a pro account, redeployed it there, and everything started working. It seems like there is a problem with my personal Vercel account that is preventing the pages from building correctly. I am going to report this through the direct help channels on Vercel to see if what was happening can be reconciled for future users.
